code/game/objects/items/documents.dm

```diff
@@ -115,3 +221,218 @@
+ * origami master. Printing at full power costs INSPECTOR_POWER_USAGE_HONK cell units
+ * instead of INSPECTOR_POWER_USAGE_NORMAL cell units.
+ */
+/obj/item/inspector/clown/bananium
+	name = "\improper Bananium HONK-spect scanner"
+	desc = "Honkmother-blessed inspection device. Performs inspections according to Clown protocols when activated, then \
+			prints a clowncrypted report regarding the maintenance of the station. Hard to replace."
+	icon_state = "bananium_inspector"
+	w_class = WEIGHT_CLASS_SMALL
+	max_mode = BANANIUM_CLOWN_INSPECTOR_PRINT_SOUND_MODE_LAST
+	///How many more times can we print?
+	var/paper_charges = 200
+	///Max value of paper_charges
+	var/max_paper_charges = 500
+	///How much charges are restored per paper consumed
+	var/charges_per_paper = 25
+
+/obj/item/inspector/clown/bananium/proc/check_settings_legality()
+	if((print_sound_mode == INSPECTOR_PRINT_SOUND_MODE_NORMAL)&&(print_time < 1 SECONDS))
+		if(cell.use(power_to_speak))
+			say("Setting combination forbidden by Geneva convention revision CCXXIII selected, reverting to defaults")
+		print_time = 5 SECONDS
+		print_sound_mode = INSPECTOR_PRINT_SOUND_MODE_CLASSIC
+		power_per_print = INSPECTOR_POWER_USAGE_NORMAL
+
+/obj/item/inspector/clown/bananium/attackby(obj/item/I, mob/user, params)
+	..()
+	check_settings_legality()
+	if(istype(I, /obj/item/paper/fake_report)||(paper_charges>=max_paper_charges))
+		to_chat(user, "<span class='info'>\The [src] refuses to consume \the [I]!</span>")
+	else if(istype(I, /obj/item/paper))
+		to_chat(user, "<span class='info'>\The [src] consumes \the [I]!</span>")
+		paper_charges+=charges_per_paper
+		if(paper_charges>max_paper_charges)
+			paper_charges = max_paper_charges
+		qdel(I)
+
+/obj/item/inspector/clown/bananium/Initialize()
+	. = ..()
+	playsound(src, 'sound/effects/angryboat.ogg', 150, FALSE)
+
+/obj/item/inspector/clown/bananium/create_slip()
+	if(print_time == 0.1 SECONDS)
+		var/obj/item/paper/fake_report/water/slip = new(get_turf(src))
+		slip.generate_report(get_area(src))
+	else
+		..()
+
+/obj/item/inspector/clown/bananium/print_report(mob/user)
+	if(print_time != 0.1 SECONDS)
+		..()
+	else if(paper_charges == 0)
+		if(cell.use(power_to_speak))
+			say("ERROR! OUT OF PAPER! MAXIMUM PRINTING SPEED UNAVAIBLE! SWITCH TO A SLOWER SPEED TO OR PROVIDE PAPER!")
+		else
+			to_chat(user, "<span class='info'>\The [src] doesn't seem to be on... Perhaps it ran out of power?")
+	else
+		paper_charges--
+		..()
+
+/obj/item/inspector/clown/bananium/cycle_print_time(mob/user)
+	var/message = "You turn the screw-like dial, setting the device's scanning speed to "
+	if(print_time == 0.1 SECONDS)
+		power_per_print = INSPECTOR_POWER_USAGE_NORMAL
+		print_time = 5 SECONDS
+		message += "SLOW."
+	else if(print_time == 5 SECONDS)
+		print_time = 1 SECONDS
+		message += "LIGHTNING FAST."
+	else
+		print_time = 0.1 SECONDS
+		power_per_print = INSPECTOR_POWER_USAGE_HONK
+		message += "HONK!"
+	balloon_alert(user, message)
```

Comment:
```suggestion
/obj/item/inspector/clown/bananium/cycle_print_time(mob/user)
	var/message
	switch(print_time)
		if(0.1 SECONDS)
			power_per_print = INSPECTOR_POWER_USAGE_NORMAL
			print_time = 5 SECONDS
			message = "SLOW."
		if(5 SECONDS)
			print_time = 1 SECONDS
			message = "LIGHTNING FAST."
		else
			print_time = 0.1 SECONDS
			power_per_print = INSPECTOR_POWER_USAGE_HONK
			message = "HONK!"
	balloon_alert(user, "You turn the screw-like dial, setting the device's scanning speed to [message]")
```

However, this should be indexed by on what mode the scanner is in, not by how long it takes to print.